docs/source/nlp/text_classification.rst

```diff
@@ -1,12 +1,169 @@
+- source_data_dir: directory of your dataset
+- target_data_dir: directory to save the converted dataset
+
+After the conversion, the TARGET_PATH should contain the following files:
+
+.. code::
+
+   .
+   |--TARGET_PATH
+     |-- train.tsv
+     |-- dev.tsv
+     |-- test.tsv
+
+Some datasets do not have the test set or their test set does not have any labels, therefore the corresponding file may be missing.
+
+Model Training
+--------------
+You may find an example of a config file to be used for training of the text classification model at `NeMo/examples/nlp/text_classification/conf/text_classification_config.yaml <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/text_classification/conf/text_classification_config.yaml>`__.
+You can change any of these parameters directly from the config file or update them with the command line arguments.
+
+The config file contains three main sections:
+- trainer: Trainer section contains the configs for PTL training and you may find more info at
+:doc:`../../introduction/core.html#model-training` and
+`PTL Trainer class API <https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#trainer-class-api>'.
+- exp_manager: the configs of experiment manager. You can find more info at :doc:`../../introduction/core.html#experiment-manager`
+- model: contains the configs of the datasets, model architecture, tokenizer, optimizer, scheduler, etc.
+
+Some sub-sections of the model section are shared among most of the NLP models. The details of these sections can be found at:
+- tokenizer and language_model: The configs of the tokenizer and encoder part and the :doc:`../nlp_model.html#important-parameters`
+- optim: the configs of the optimizer and scheduler :doc:`../../introduction/core.html`
+
+Example of a command for training a text classification model on two GPUs for 50 epochs:
+
+.. code::
+
+    python examples/nlp/text_classification/text_classification_with_bert.py \
+        model.training_ds.file_path=<TRAIN_FILE_PATH> \
+        model.validation_ds.file_path=<VALIDATION_FILE_PATH> \
+        trainer.max_epochs=50 \
+        trainer.gpus=[0,1] \
+        optim.name=adam \
+        optim.lr=0.0001 \
+        model.nemo_path=<NEMO_FILE_PATH>
+
+
+By default, the final model after training is saved in the path specified by 'NEMO_FILE_PATH'.
+
+Model Arguments
+^^^^^^^^^^^^^^^
+The following table lists some of the model's parameters you may use in the config files and set them from command line when training a model:
+
++-------------------------------------------+-----------------+------------------------------------------------+--------------------------------------------------------------------------------------------------------------+
+| **Parameter**                             | **Data Type**   |   **Default**                                  | **Description**                                                                                              |
```

Comment:
you might want to drop the defaults as all these values are in the referenced config

Reply:
Actually these defaults should be different from the config file. They should be the values if not set by the config file. In most cases, they are the same, but not always. Currently these values are the defaults of the config file. How about putting the defaults if they are not set in the config?